packages/sommelier.rb

```diff
@@ -3,112 +3,172 @@
+
+        # point virtwl path to /dev/null (virtwl path is hardcoded now)
+        sed -i -e 's:#define VIRTWL_DEVICE "/dev/wl0":#define VIRTWL_DEVICE "/dev/null":' virtualization/virtwl_channel.cc
+      SED
+      
+      # replace the virtwl shm driver to noop shm driver (removed in commit 180e42d)
+      @sommelier_shm_patch = { /^[[:blank:]]+struct sl_host_buffer\*.+?host_buffer->resource;$/m => <<~P1.chomp, /^[[:blank:]]+host_shm_pool->fd = fd;$/ => <<~P2.chomp }
+        assert(host->proxy);
+        sl_create_host_buffer(host->shm->ctx, client, id,
+            wl_shm_pool_create_buffer(host->proxy, offset, width, height, stride, format), width, height);
+      P1
+        host_shm_pool->proxy = wl_shm_create_pool(host->shm_proxy, fd, size);
+        wl_shm_pool_set_user_data(host_shm_pool->proxy, host_shm_pool);
+        close(fd);
+      P2
+
+      @sommelier_shm = File.read('sommelier-shm.cc')
+      @sommelier_shm_patch.each_pair do |k, v|
+        @sommelier_shm.sub!(k, v)
+      end
+
+      File.write('sommelier-shm.cc', @sommelier_shm)
+    end
   end
 
   def self.build
     Dir.chdir('vm_tools/sommelier') do
       # lld is needed so libraries linked to system libraries (e.g. libgbm.so) can be linked against, since those are required for graphics acceleration.
-
       system <<~BUILD
-        env CC=clang CXX=clang++ \
+        #{CREW_ENV_OPTIONS.gsub('gold', 'lld')} CC=clang CXX=clang++ \
           meson #{CREW_MESON_OPTIONS} \
           -Db_asneeded=false \
           -Db_lto=true \
           -Db_lto_mode=thin \
           -Dxwayland_path=#{CREW_PREFIX}/bin/Xwayland \
-          -Dxwayland_gl_driver_path=/usr/#{ARCH_LIB}/dri -Ddefault_library=both \
-          -Dxwayland_shm_driver=noop -Dshm_driver=noop -Dvirtwl_device=/dev/null \
-          -Dpeer_cmd_prefix="#{CREW_PREFIX}#{@peer_cmd_prefix}" \
+          -Dxwayland_gl_driver_path=/usr/#{ARCH_LIB}/dri \
+          -Ddefault_library=both \
+          -Dwith_tests=false \
           builddir
       BUILD
 
       system 'meson configure builddir'
       system 'samu -C builddir'
 
       Dir.chdir('builddir') do
-        system 'curl -L "https://chromium.googlesource.com/chromiumos/containers/sommelier/+/fbdefff6230026ac333eac0924d71cf824e6ecd8/sommelierrc?format=TEXT" | base64 --decode > sommelierrc'
+        @help = <<~EOF
+          To start the sommelier daemon, run 'startsommelier'
+          To stop the sommelier daemon, run 'stopsommelier'
+          To restart the sommelier daemon, run 'restartsommelier'
+        EOF
 
         @sommelierenv = <<~EOF
           set -a
-          CLUTTER_BACKEND=wayland
+
           DISPLAY=:0
-          GDK_BACKEND=x11
-          SCALE=1
           SOMMELIER_ACCELERATORS='Super_L,<Alt>bracketleft,<Alt>bracketright'
-          WAYLAND_DISPLAY=wayland-0
-          XDG_RUNTIME_DIR=/var/run/chrome
+          WAYLAND_DISPLAY=wayland-1
+
+          : "${CLUTTER_BACKEND:=x11}"
```

Comment:
Why switch this from `wayland` to `x11`?

Reply:
Because some apps might not be ready to support Wayland